mage_ai/api/resources/PipelineRunResource.py

```diff
@@ -108,26 +140,34 @@ def collection(self, query_arg, meta, user, **kwargs):
     @classmethod
     @safe_db_query
     async def process_collection(self, query_arg, meta, user, **kwargs):
-        total_results = self.collection(query_arg, meta, user, **kwargs)
+        total_results = await self.collection(query_arg, meta, user, **kwargs)
         total_count = total_results.count()
 
         limit = int(meta.get(META_KEY_LIMIT, self.DEFAULT_LIMIT))
         offset = int(meta.get(META_KEY_OFFSET, 0))
 
+        include_pipeline_uuids = query_arg.get('include_pipeline_uuids', [False])
+        if include_pipeline_uuids:
+            include_pipeline_uuids = include_pipeline_uuids[0]
+
         pipeline_type = query_arg.get('pipeline_type', [None])
         if pipeline_type:
             pipeline_type = pipeline_type[0]
 
-        if pipeline_type is not None:
-            pipeline_type_by_pipeline_uuid = dict()
+        filter_by_pipeline_type = pipeline_type is not None
+        if filter_by_pipeline_type:
             try:
+                pipeline_type_by_pipeline_uuid = dict()
                 pipeline_runs = total_results.all()
                 results = []
                 for run in pipeline_runs:
-                    if run.pipeline_uuid not in pipeline_type_by_pipeline_uuid:
-                        pipeline_type_by_pipeline_uuid[run.pipeline_uuid] = run.pipeline_type
-                    run_pipeline_type = pipeline_type_by_pipeline_uuid[run.pipeline_uuid]
-                    if run_pipeline_type == pipeline_type:
+                    filters = []
+                    if filter_by_pipeline_type:
```

Comment:
You already have a `if filter_by_pipeline_type` above. It's a duplicate check here.